independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java

```diff
@@ -23,19 +23,43 @@
     private static final String SETTINGS_GRADLE_PATH = "settings.gradle";
     private static final String GRADLE_PROPERTIES_PATH = "gradle.properties";
 
+    private final ProjectWriter rootWriter;
+
     private Model model;
 
     public GradleBuildFile(ProjectWriter writer) {
         super(writer, BuildTool.GRADLE);
+        rootWriter = writer;
+    }
+
+    public GradleBuildFile(ProjectWriter writer, ProjectWriter rootWriter) {
```

Comment:
It seems that you don't need two ProjectWriters here, but two `GradleBuildFile` instances in QuarkusAddExtension? I'll review more carefully next week.

Reply:
I don't think so. Or you have to make the AddExtensions more complex. Because you're reading the different folder based on whether it's multi-module to init the model and writing the different folder when it's closed.

Reply:
I may be missing something but I'd expect each `GradleBuildFile` to be responsible for maintaining a single `build.gradle`. Perhaps `GradleBuildFile` should have a `GradleBuildFile getParent()` method instead. 

Not saying your solution is wrong, just that it's confusing to have a single `GradleBuildFile` changing two files internally

Reply:
- Single Module
project/build.gradle
project/gradle.properties
project/settings.gradle

- Multi Module (required properties from parent or no required properties in both gradle.properties)
sub/build.gradle
parent/gradle.properties
parent/settings.gradle

- Multi Module (required properties from sub)
sub/build.gradle
sub/gradle.properties
parent/settings.gradle

I think three files combined is a GradleBuildFile.

Reply:
There's only one file for one type. It does not dealing with both.